File: 240128/체크판위에서 2/on-the-checkboard-2.py
R,C=map(int,input().split())
graph=[list(map(str,input().split())) for _ in range(R)]

ans=0

for i in range(1,R-2):
    for j in range(1,C-2):
        for k in range(i+1,R-1):
            for l in range(j+1,C-1):
                if graph[0][0]!=graph[i][j] and \
                   graph[i][j]!=graph[k][l] and \
                   graph[k][l]!=graph[R-1][C-1]:
                        ans+=1

print(ans)

File: 260106/체크판위에서 2/on-the-checkboard-2.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const [r, c] = input[0].split(' ').map(Number);
const grid = input.slice(1).map(line => line.trim().split(' '));

// Please Write your code here.
let ans =0;
for(let i=1; i<r-2; i++){
    for(let j=1; j<c-2; j++){
        for(let k=i+1; k<r-1; k++){
            for(let l=j+1; l<c-1; l++){
                if(grid[i][j]!==grid[0][0]
                && grid[k][l]!==grid[r-1][c-1]
                &&grid[i][j]!==grid[k][l]){
                    ans++;
                }
            }
        }
    }
}
console.log(ans)